### PR TITLE
Add support for JSON logging format

### PR DESCRIPTION
**Pull request by erikgb**
In this PR I propose to introduce slog.Handler as a backend for logr (and klog) - as documented [here](https://github.com/go-logr/logr?tab=readme-ov-file#using-a-sloghandler-as-backend-for-logr).

I am doing a minimal change in this PR where my primary goal is to add support for a structured log format (ref.  https://github.com/cert-manager/trust-manager/issues/282).

**NOTE**: This might break for some users, depending on if/how they parse trust-manager log output. I cannot imagine the slog text format (the new default) is the same as the deprecated klog format. We always try to avoid breaking changes, but I think a change is required here anyway - since the currently used klog format is deprecated.

Fixes https://github.com/cert-manager/trust-manager/issues/282

### CODE REVIEW

File: cmd/trust-manager/app/options/options.go
Comment (erikgb) [inline cmd/trust-manager/app/options/options.go:82]:
```suggestion
		return string(logFormatText)
```

**Review by hawksight — APPROVED** [review]
Had a bit of a read through of the PR and of the linked documentation. This seems to make sense to me. The notion for using slog is that this is in the Go standard library now? I can't approve and can't comment explicitly on the code, but if it works 👍 

Can I run this? just need to branch checkout and build the image?

**Comment by erikgb** [comment]
> Had a bit of a read through of the PR and of the linked documentation. This seems to make sense to me. The notion for using slog is that this is in the Go standard library now? I can't approve and can't comment explicitly on the code, but if it works 👍
> 
> Can I run this? just need to branch checkout and build the image?

Thanks @hawksight! You should be able to just checkout the branch and install it. My workstation environment is really borked for working with open source.

File: cmd/trust-manager/app/options/options.go
Comment (SgtCoDFish) [inline cmd/trust-manager/app/options/options.go:87]: comment: Unrelated to this PR but I'd never actually stopped to consider how bonkers it is that `flag.Set("v", o.logLevel)` is how a verbosity level _should_ be set according to klogr!

File: cmd/trust-manager/app/options/options.go
Comment (SgtCoDFish) [inline cmd/trust-manager/app/options/options.go:220]: suggestion (non-blocking): I think changing the type of this arg is a breaking change.

Currently with the latest release of trust-manager I can pass `--log-level=foo5` and it'll silently ignore the invalid number and default to log level 1.

On this branch, if I pass the same arg it fails:

```text
error: invalid argument "boo1" for "-v, --log-level" flag: strconv.ParseInt: parsing "foo1": invalid syntax
```

I think one option to avoid the break would be to keep this as a string and add `func (lo *logOptions) LogLevel() (int, error)` for getting the log level, so we could handle the error ourselves (and probably just default to log level 1 and log a non-fatal error). I don't think that's required but I think it'd be nice. Your call!

**Review by SgtCoDFish — APPROVED** [review]
This looks great, thank you for doing it!

I've added a suggestion but it's not a blocker. Adding approval with a hold in case you want to address the suggestion!

Thanks also to @hawksight for testing this yesterday!

/lgtm
/approve
/hold

File: cmd/trust-manager/app/options/options.go
Comment (erikgb) [reply cmd/trust-manager/app/options/options.go:87]: Agree, but that's history after this PR. 😉

File: cmd/trust-manager/app/options/options.go
Comment (erikgb) [reply cmd/trust-manager/app/options/options.go:220]: Thanks! Good catch! I will revert the change of type.

Comment (erikgb) [reply cmd/trust-manager/app/options/options.go]: Fixed now, but I think we should accept this breaking change. Who would ever set `--log-level=foo5`? 😉 I consider changing the type as a bugfix. Was the existing behavior (accepting any string) intentional?

Comment (erikgb) [reply cmd/trust-manager/app/options/options.go:220]: This was discussed in todays stand-up, and we reached a consensus of considering the switch of `log-level` CLI arg type from string to int a bugfix.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve
/unhold

I really like this, and I think if we document the change from string to int for the log level then that's totally reasonable 👍 

Thanks for doing this Erik!

**Pull request merged**
